jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/util/SLAComplianceCell.java

```diff
@@ -0,0 +1,113 @@
+
+import com.google.gwt.cell.client.AbstractCell;
+import com.google.gwt.safehtml.shared.SafeHtmlBuilder;
+
+import elemental2.dom.DOMTokenList;
+import elemental2.dom.HTMLElement;
+
+@Dependent
+@Templated(value = "SLAComplianceCell.html")
+public class SLAComplianceCell extends AbstractCell<Integer> implements IsElement {
+
+    private static final String SPAN_NAME = "sla-compliance";
+    private static final String SLA_STYLE = "label";
+    private static final String SLA_NA_PRESENT_STYLE = "label-default";
+    private static final String SLA_PENDING_PRESENT_STYLE = "label-primary";
+    private static final String SLA_MET_PRESENT_STYLE = "label-success";
+    private static final String SLA_ABORTED_PRESENT_STYLE = "label-warning";
+    private static final String SLA_VIOLATED_PRESENT_STYLE = "label-danger";
+    
+    private List<String> descriptions;
+    
+    @Inject
+    @DataField(SPAN_NAME)
+    @Named("span") 
+    private HTMLElement content;
+    
+    public SLAComplianceCell init(final List<String> descriptions) {
+        this.descriptions = descriptions;
+        return this;
+    }
+
+    @Override
+    public void render(Context context,
+                       Integer value,
+                       SafeHtmlBuilder sb) {
+
+        DOMTokenList tooltipClasses = content.classList;
+        
+        tooltipClasses.remove(SLA_NA_PRESENT_STYLE);
+        tooltipClasses.remove(SLA_PENDING_PRESENT_STYLE);
+        tooltipClasses.remove(SLA_MET_PRESENT_STYLE);
+        tooltipClasses.remove(SLA_ABORTED_PRESENT_STYLE);
+        tooltipClasses.remove(SLA_VIOLATED_PRESENT_STYLE);
+        
+        tooltipClasses.add(SLA_STYLE);
+        String description = descriptions.get(0);
+        switch (value) {
+            case ProcessInstance.SLA_NA:
+                description = descriptions.get(1);
+                tooltipClasses.add(SLA_NA_PRESENT_STYLE);
+                
+                break;
+            case ProcessInstance.SLA_PENDING:
+                description = descriptions.get(2);
+                tooltipClasses.add(SLA_PENDING_PRESENT_STYLE);  
+                break;
+            case ProcessInstance.SLA_MET:
+                description = descriptions.get(3);
+                tooltipClasses.add(SLA_MET_PRESENT_STYLE);
+                break;
+            case ProcessInstance.SLA_ABORTED:
+                description = descriptions.get(4);
+                tooltipClasses.add(SLA_ABORTED_PRESENT_STYLE);
+                break;
+            case ProcessInstance.SLA_VIOLATED:
+                description = descriptions.get(5);
+                tooltipClasses.add(SLA_VIOLATED_PRESENT_STYLE);               
+                break;
+            default:
+                description = descriptions.get(0);
+                tooltipClasses.add(SLA_NA_PRESENT_STYLE);
+                break;
+
+        }        
+        content.textContent = description; 
+        sb.appendHtmlConstant(content.innerHTML);
```

Comment:
@cristianonicolai @mbarkley how to render complete HTMLElement here? There is no outerHtml thingy in elemental2.... so with this refactored code there are no coloured labels any more which pretty much makes this custom cell useless...

Reply:
@mswiderski seems like outerHTML is actually missing in the API: https://github.com/google/elemental2/issues/37

I think in this case we could simplify things and just wrap with another element, it also seems that using Errai template would be unnecessary and actually share the element across multiple render calls, have a look in my suggestions here: https://gist.github.com/cristianonicolai/59d6411865dd1b1610cc2f66a315b2eb

Reply:
@cristianonicolai thanks a lot, this works perfectly. all good I believe, do you mind to take a look again?